File: 979. Distribute Coins in Binary Tree.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def distributeCoins(self, root: Optional[TreeNode]) -> int:
        ans=0
        def rec(node) :
            nonlocal ans
            if not node :
                return 0
            l=rec(node.left)
            r=rec(node.right)
            ans += abs(l) + abs(r)
            return node.val + l + r - 1
        rec(root)
        return ans

File: 979. Distribute Coins in Binary Tree.js
/**
 * Definition for a binary tree node.
 * function TreeNode(val, left, right) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.left = (left===undefined ? null : left)
 *     this.right = (right===undefined ? null : right)
 * }
 */
/**
 * @param {TreeNode} root
 * @return {number}
 */
var distributeCoins = function(root) {
    let ans=0;
    const func=(node)=>{
        if(node==null)
        {
            return 0;
        }
        let l=func(node.left);
        let r=func(node.right);
        ans += Math.abs(l) + Math.abs(r);
        return node.val + l + r -1;
    }
    func(root);
    return ans
};
